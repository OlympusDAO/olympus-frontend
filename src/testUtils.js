import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import { light as lightTheme } from "./themes/light.js";
import { Web3ContextProvider } from "./hooks/web3Context";
import store from "./store.ts";
import { createMemoryHistory } from "history";
import App from "./App";
import { I18nProvider } from "@lingui/react";
import { i18n } from "@lingui/core";

const AllTheProviders = ({ children }) => {
  return (
    <Web3ContextProvider>
      <Provider store={store}>
        <BrowserRouter basename={"/"}>
          <ThemeProvider theme={lightTheme}>
            <CssBaseline />
            {children}
          </ThemeProvider>
        </BrowserRouter>
      </Provider>
    </Web3ContextProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

const renderRoute = function (route) {
  const history = createMemoryHistory();
  history.push(route);
  render(
    <Web3ContextProvider>
      <Provider store={store}>
        <I18nProvider i18n={i18n}>
          <BrowserRouter basename={"/#"}>
            <App />
          </BrowserRouter>
        </I18nProvider>
      </Provider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
