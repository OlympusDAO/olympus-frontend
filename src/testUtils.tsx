import { i18n } from "@lingui/core";
import { I18nProvider } from "@lingui/react";
import CssBaseline from "@material-ui/core/CssBaseline";
import { ThemeProvider } from "@material-ui/core/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React, { ReactElement, ReactNode } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

import App from "./App";
import { Web3ContextProvider } from "./hooks/web3Context";
import { ReactQueryProvider } from "./lib/react-query";
import defaultStore from "./store";
import { light as lightTheme } from "./themes/light.js";

const customRender = (ui: ReactElement, store = defaultStore, options?: RenderOptions): RenderResult => {
  const ProviderWrapper = ({ children }: { children?: ReactNode }) => (
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <ThemeProvider theme={lightTheme}>
                <CssBaseline />
                {children}
              </ThemeProvider>
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>
  );
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const renderRoute = function (route: string, store = defaultStore) {
  const history = createMemoryHistory();
  history.push(route);
  return render(
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <App />
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
