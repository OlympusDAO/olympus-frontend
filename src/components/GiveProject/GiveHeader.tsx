import { Trans } from "@lingui/macro";
import { Box, Button, Link, Typography } from "@material-ui/core";
import { BigNumber } from "bignumber.js";
import { NavLink, useLocation } from "react-router-dom";
import { NetworkId } from "src/constants";
import { isSupportedChain } from "src/slices/GiveThunk";

type GiveHeaderProps = {
  isSmallScreen: boolean;
  isVerySmallScreen: boolean;
  totalDebt: BigNumber;
  networkId: NetworkId;
};

export function GiveHeader({ isSmallScreen, isVerySmallScreen, totalDebt, networkId }: GiveHeaderProps) {
  const location = useLocation();
  const isDonationsTabActive = location.pathname.replace("/", "") == "give/donations";
  const isGiveTabActive =
    location.pathname.replace("/", "") == "give" || location.pathname.replace("/", "").includes("give/projects");

  return (
    <Box className={`give-subnav ${isSmallScreen || isVerySmallScreen ? "smaller" : ""}`}>
      <Link
        component={NavLink}
        id="give-sub-dash"
        to="/give"
        className={`give-option ${isGiveTabActive ? "give-active" : ""}`}
      >
        <Button color="secondary">
          <Typography variant="h6">
            <Trans>Give</Trans>
          </Typography>
        </Button>
      </Link>
      <Link
        component={NavLink}
        id="give-sub-donations"
        to="/give/donations"
        className={`give-option ${isDonationsTabActive ? "give-active" : ""}`}
      >
        <Button color="secondary">
          <Typography variant="h6">
            <Trans>My Donations</Trans>
          </Typography>
        </Button>
      </Link>
      {new BigNumber(totalDebt).gt(new BigNumber(0)) && isSupportedChain(networkId) ? (
        <Link
          component={NavLink}
          id="give-sub-redeem"
          to="/give/redeem"
          className={`give-option ${location.pathname.replace("/", "") == "give/redeem" ? "give-active" : ""}`}
        >
          <Button color="secondary">
            <Typography variant="h6">
              <Trans>Redeem</Trans>
            </Typography>
          </Button>
        </Link>
      ) : (
        <></>
      )}
    </Box>
  );
}
