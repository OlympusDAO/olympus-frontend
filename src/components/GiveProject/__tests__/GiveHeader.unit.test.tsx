import { BigNumber } from "bignumber.js";

import { render } from "../../../testUtils";
import { GiveHeader } from "../GiveHeader";

describe("<GiveHeader/>", () => {
  it("should render component", () => {
    const { container } = render(
      <GiveHeader isSmallScreen={false} isVerySmallScreen={false} totalDebt={new BigNumber(123123123)} networkId={1} />,
    );
    expect(container).toMatchSnapshot();
  });
});
