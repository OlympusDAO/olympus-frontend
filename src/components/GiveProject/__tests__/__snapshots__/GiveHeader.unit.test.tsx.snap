// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GiveHeader/> should render component 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-1 give-subnav "
  >
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineNone give-option  MuiTypography-colorPrimary"
      href="/#/give"
      id="give-sub-dash"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-disableElevation"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            Give
          </h6>
        </span>
      </button>
    </a>
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineNone give-option  MuiTypography-colorPrimary"
      href="/#/give/donations"
      id="give-sub-donations"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-disableElevation"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            My Donations
          </h6>
        </span>
      </button>
    </a>
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineNone give-option  MuiTypography-colorPrimary"
      href="/#/give/redeem"
      id="give-sub-redeem"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSecondary MuiButton-disableElevation"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            Redeem
          </h6>
        </span>
      </button>
    </a>
  </div>
</div>
`;
